components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/LocalAuthenticatorAdapter.java

```diff
@@ -33,5 +34,8 @@ public LocalAuthenticatorAdapter(LocalAuthenticatorConfig config) {
 
         authenticatorName = config.getName();
         friendlyName = config.getDisplayName();
+        if (config instanceof UserDefinedLocalAuthenticatorConfig) {
+            authenticationType = ((UserDefinedLocalAuthenticatorConfig) config).getAuthenticationType();
```

Comment:
as this adapter is for user defined authenticators can'r we expect the exact type from the constructor. So this casting is not needed